### PR TITLE
Update dropout layer to be compatible with TF1 and TF2.

### PR DESCRIPTION
**Pull request by jarednielsen**
The function signature for tf.nn.dropout in TF1 is:
```python
tf.nn.dropout(
    x,
    keep_prob=None,
    noise_shape=None,
    seed=None,
    name=None,
    rate=None
)
```

while TF2 has:
```python
tf.nn.dropout(
    x, rate, noise_shape=None, seed=None, name=None
)
```
Using the `rate` keyword argument is compatible with both, and less confusing.

### CODE REVIEW

**Review by shahminfikri — APPROVED** [review]

**Comment by jarednielsen** [comment]
Perfect, just needs to be merged in.
